bootkube/resources/charts/calico/templates/host-protection.yaml

```diff
@@ -0,0 +1,108 @@
+{{- if .Values.calico.hostProtection.enabled }}
+# General
+# =======
+---
+# Disable default failsafe inbound rules
+apiVersion: crd.projectcalico.org/v1
+kind: FelixConfiguration
+metadata:
+  name: default
+spec:
+  failsafeInboundHostPorts: []
+
+# Firewall policy
+# ===============
+---
+# SSH access
+apiVersion: crd.projectcalico.org/v1
+kind: GlobalNetworkPolicy
+metadata:
+  name: ssh
+spec:
+  selector: has(host-endpoint)
+  preDNAT: true
+  applyOnForward: true
+  order: 0
+  ingress:
+  - action: Allow
+    protocol: TCP
+    source:
+      nets:
+      {{- range .Values.calico.hostProtection.managementCIDRs }}
+      - {{ . }}
+      {{- end }}
+    destination:
+      ports: [22]
+---
+# kubectl access
+apiVersion: crd.projectcalico.org/v1
+kind: GlobalNetworkPolicy
+metadata:
+  name: kubectl
+spec:
+  selector: nodetype == 'controller'
+  preDNAT: true
+  applyOnForward: true
+  order: 0
+  ingress:
+  - action: Allow
+    protocol: TCP
+    destination:
+      ports: [6443]
+---
+# Traffic between cluster nodes
+apiVersion: crd.projectcalico.org/v1
+kind: GlobalNetworkPolicy
+metadata:
+  name: inter-node-traffic
+spec:
+  selector: has(host-endpoint)
+  preDNAT: true
+  applyOnForward: true
+  order: 10
+  ingress:
+    - action: Allow
+      source:
+        nets:
+        - {{ .Values.calico.hostProtection.nodeCIDR }}
+---
+# Ingress traffic to Contour
+apiVersion: crd.projectcalico.org/v1
+kind: GlobalNetworkPolicy
+metadata:
+  name: contour-ingress
+spec:
+  selector: nodetype == 'worker'
+  preDNAT: true
+  applyOnForward: true
+  order: 20
+  ingress:
+  - action: Allow
+    protocol: TCP
+    destination:
+      ports: [80, 443]
```

Comment:
What happens on upgrade if the user has modified this global network policy? (just curious)

Reply:
The change will be reverted. All changes should be done via `helm`.

Reply:
:ok_hand: